internal/statemachine/classic_states.go

```diff
@@ -275,6 +277,54 @@ func (stateMachine *StateMachine) extractRootfsTar() error {
 	return nil
 }
 
+// germinate runs the germinate binary and parses the output to create
+// a list of packages from the seed section of the image definition
+func (stateMachine *StateMachine) germinate() error {
+	var classicStateMachine *ClassicStateMachine
+	classicStateMachine = stateMachine.parent.(*ClassicStateMachine)
+
+	// create a scratch directory to run germinate in
+	germinateDir := filepath.Join(classicStateMachine.stateMachineFlags.WorkDir, "germinate")
+	err := osMkdir(germinateDir, 0755)
+	if err != nil && !os.IsExist(err) {
+		return fmt.Errorf("Error creating germinate directory: \"%s\"", err.Error())
+	}
+
+	germinateCmd := generateGerminateCmd(classicStateMachine.ImageDef)
+	germinateCmd.Dir = germinateDir
+
+	if germinateOutput, err := germinateCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("Error running germinate command \"%s\". Error is \"%s\". Output is: \n%s",
+			germinateCmd.String(), err.Error(), string(germinateOutput))
+	}
+
+	packageMap := make(map[string]*[]string)
+	packageMap[".seed"] = &classicStateMachine.Packages
+	packageMap[".snaps"] = &classicStateMachine.Snaps
+	for fileExtension, packageList := range packageMap {
+		for _, fileName := range classicStateMachine.ImageDef.Rootfs.Seed.Names {
+			seedFilePath := filepath.Join(germinateDir, fileName+fileExtension)
+			seedFile, err := osOpen(seedFilePath)
+			if err != nil {
+				return fmt.Errorf("Error opening seed file %s: \"%s\"", seedFilePath, err.Error())
+			}
+			defer seedFile.Close()
+
+			seedScanner := bufio.NewScanner(seedFile)
+			for seedScanner.Scan() {
+				seedLine := seedScanner.Bytes()
+				matched, _ := regexp.Match(`^[a-z0-9].*`, seedLine)
```

Comment:
If I understand correctly, this regexp makes sure we only pick up actual packages from the `.seed`/`.snap` files, right? Wonder if we could do it somehow faster, maybe putting assumptions somewhere else: like starting the read from the third line (since we know two-first are label and border) and then reading till the next border. Or checking if first letter is lowercase and `!= "-"`.
...on the other hand, I guess I'm over-complicating stuff. We live in times where regular expressions are much faster than in the past, so this should be good as-is.